pull_chart_tarball.go

```diff
@@ -52,7 +52,7 @@ func (c *Client) doFile(ctx context.Context, req *http.Request) (string, error)
 			// Github pages 404 produces full HTML page which
 			// obscures the logs.
 			if resp.StatusCode == http.StatusNotFound {
-				return backoff.Permanent(microerror.Maskf(executionFailedError, fmt.Sprintf("got StatusCode %d for url %#q", resp.StatusCode, req.URL.String())))
+				return backoff.Permanent(microerror.Maskf(pullChartFailedError, fmt.Sprintf("got StatusCode %d for url %#q", resp.StatusCode, req.URL.String())))
```

Comment:
Why do we want to treat these two cases the same way? Isn't there any value in differentiating them? "Not Found" is a pretty different root cause than "No Such Host". Why ruling them all the same way?

Reply:
There's no serious reason against differentiating it, the only reason was that I don't want to have multiple `if else` statement in `chart-operator` when matcher function in using.

Reply:
I would return `notFound` error in this case and return `pullChartURLError` for other cases, WDYT?

Reply:
I'm fine with `pullChartFailedError` for both.

We need to pull the chart tarball over HTTP and it doesn't work. That is what we care about. The reason why is clear from the error text.